### PR TITLE
Add IAddChild support

### PR DESCRIPTION
**Pull request by Gillibald**
## Checklist

- [x] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0021537-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

**Comment by pr8x** [comment]
May I ask what this is about 😅?

**Comment by maxkatz6** [comment]
A forgotten, but useful, WPF API
https://docs.microsoft.com/en-us/dotnet/api/system.windows.markup.iaddchild?view=windowsdesktop-6.0

File: src/Markup/Avalonia.Markup.Xaml/IAddChild.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Avalonia.Markup.Xaml
+{
+    public interface IAddChild
```
Comment (kekekeks) [inline src/Markup/Avalonia.Markup.Xaml/IAddChild.cs:3]: That was probably a wrong place for the interface.

Comment (Gillibald) [reply src/Markup/Avalonia.Markup.Xaml/IAddChild.cs:3]: Any suggestions?

Comment (kekekeks) [reply src/Markup/Avalonia.Markup.Xaml/IAddChild.cs:3]: Well, `Avalonia.Markup.Xaml` references both `Avalonia.Controls` and `Avalonia.Base`. You probably want to implement `IAddChild` on `TextBlock` from `Avalonia.Controls`, so it should at least be in `Avalonia.Controls`.
Our `[Content]`, `[DataType]`, `[TemplateContent]` and several other attributes live in `Avalonia.Base`, so we could have `IAddChild` there as well

**Pull request merged**